src/cljs/athens/db.cljs

```diff
@@ -417,17 +413,16 @@
   ([query n ignore-dup]
    (if (string/blank? query)
      (vector)
-     (let [results (->> (d/q '[:find [(pull ?node [:db/id :node/title :block/uid]) ...]
-                               :in $ ?query-pattern ?query
-                               :where
-                               [?node :node/title ?title]
-                               [(re-find ?query-pattern ?title)]
-                               [(not= ?title ?query)]]                ;; ignore exact match to avoid duplicate
-                             @dsdb
-                             (re-case-insensitive query)
-                             (when ignore-dup query))
-                        (take n))]
-       results))))
+     (let [ignore-dup             (when ignore-dup query)
+           case-insensitive-query (re-case-insensitive query)]
+       (sequence
+         (comp
+           (filter #(re-find case-insensitive-query (:v %)))
+           (filter #(not= (:v %) ignore-dup))
+           (take n)
+           (map #(:e %))
+           (map #(d/entity @dsdb %)))
+         (d/datoms @dsdb :aevt :node/title))))))
```

Comment:
Use ->> to rewrite it in more readable form
(->> (d/datoms) .. 
         (map (comp :e #(d/entity ..))
         (take n)
         (filter (apply every-pred(https://clojuredocs.org/clojure.core/every-pred) filter1 filter2))
         seq)

Reply:
@arkRedM

1. This is a transducer that's why I use `sequence` and `comp`. Same as https://github.com/athensresearch/athens/blob/95ca61c4c1ea5554b49a7c196ad1540cc8bea1d2/src/cljs/athens/db.cljs#L446 and https://github.com/athensresearch/athens/blob/95ca61c4c1ea5554b49a7c196ad1540cc8bea1d2/src/cljs/athens/db.cljs#L452
2. Thanks for pointing out the use of `every-pred`. I finally realized its use case.

Reply:
I agree this is the use case for transducers, but in terms of readability and what is happening(for someone who is looking at it) will find it easier if all steps are laid out instead of hiding some complexity inside a transducer(makes it easier to debug).

It's totally fine by me to keep it as is, but see if one is more readable than other.

Reply:
@arkRedM Another thing to consider is that `transducers`, at least from what I read, is faster https://groups.google.com/g/clojure/c/iIju063QmG0/m/GKzD_SpvBgAJ

And also, `comp` is read like `->>` when used in transducers. https://clojure.org/reference/transducers#_defining_transformations_with_transducers

But all this is theory and you have more experience. :) Let me know your final thoughts on this. I'll update this to whatever is more appropriate.

Reply:
Should be pretty easy to `time` performance between using transducer vs `->>` right? The Google Groups link references CLJ, not CLJS so maybe difference. Either way, I haven't used transducers yet and this is way faster so I love it.